DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/svd/src/non_std_convariance.hpp

```diff
@@ -0,0 +1,207 @@
+#ifndef __NON_STD_COVARIANCE_MATRIX_HPP__
+#define __NON_STD_COVARIANCE_MATRIX_HPP__
+
+namespace fpga_linalg {
+// This functor computes the columns x columns covariance matrix of a rows x
+// columns input matrix A
+
+// The deign compute T tah is is transpose(A)*A 
+
+template <typename T,          // The datatype for the computation
+          unsigned rows,       // Number of rows in the A matrices
+          unsigned columns,    // Number of columns in the A matrices
+          unsigned pipe_size,  // Number of elements read/write per pipe
```

Comment:
```suggestion
          unsigned kRows,       // Number of rows in the A matrices
          unsigned kColumns,    // Number of columns in the A matrices
          unsigned kPipeSize,  // Number of elements read/write per pipe
```
Variable names should be `snake_case`
Function names should be `PascalCase()`
constants should be `kCamelCase` (the leading `k` is important)